libraries/AP_OpticalFlow/AP_OpticalFlow_UPFLOW.cpp

```diff
@@ -87,6 +259,91 @@ void AP_OpticalFlow_UPFLOW::init()
+        |     0x96     |                 | status code:  0x00 == success
+        |     0x26     |   status code   |               0x01 == IIC no response
+        |     0xbc     |                 |               0x02 == XOR error
+        |     0x50     |                 | send: 0xAB, 0x96, 0x26, 0xbc, 0x50, 0x5c(BCC without 0xAB)
+        |   BCC(0x5c)  | 0xAB xor status | success response: 0xAB, 0x00, 0xAB
+        | ------------ | --------------- |
+        |     0xBB     |      0xBB       | 3. sensor parameter configuration
+        |     0xdc     |                 |    sending data in upflow_sensor_cfg
+        |    address   |   status code   | same status code above
+        |     data     |                 | send: 0xBB, 0xdc, 0xXX, 0xXX, 0xXX(BCC without 0xBB)
+        |     BCC      | 0xBB xor status | success response: 0xBB, 0x00, 0xBB
+        | ------------ | --------------- |
+        |     0xDD     |      none       | 4. Stop flag
+
+        !:  Step 3 will run repeatedly until the sending is complete.
+            Each time the module receives an 0xBB instruction, it will return a corresponding status.
+    */
+
+    // uint32_t nbytes = 0;
+    uint8_t bcc_sum = 0;
+    uint8_t recv_buf[3];
+
+    //delay 100ms before initialization
+    hal.scheduler->delay(100);
+
+    // GCS_SEND_TEXT(MAV_SEVERITY_INFO, "UPFLOW: Start initialization!");
+
+    // (0xAA)start configuration
+    uart->write((uint8_t)0xAA);
+
+    // (0xAB)internal parameter configuration
+    uart->write((uint8_t)0xAB);
+    for (int i = 0; i < 4; i++) {
+        uart->write(upflow_internal_para[i]);
+        bcc_sum ^= upflow_internal_para[i];
+    }
+    uart->write(bcc_sum);
+    hal.scheduler->delay(10);
+    for (int i = 0; i < 3; i++) {
+        recv_buf[i] = uart->read();
+    }
+    // if failed to receive response code, you can start over from 0xAA.
+    if ((recv_buf[0] ^ recv_buf[1]) != recv_buf[2]) {
+        UPFLOW_INIT_FAILED;
+        return;
+    }
+
+    // (0xBB)sensor parameter configuration
+    uint32_t cfg_cnt = 0;
+    while (cfg_cnt < sizeof(upflow_sensor_cfg) - 1) {
+        uart->write((uint8_t)0xBB);
+        uart->write(UPFLOW_SENSOR_IIC_ADDR);
+        uart->write(upflow_sensor_cfg[cfg_cnt]); // address
+        uart->write(upflow_sensor_cfg[cfg_cnt + 1]); // data
+        uart->write(UPFLOW_SENSOR_IIC_ADDR ^ upflow_sensor_cfg[cfg_cnt] ^ upflow_sensor_cfg[cfg_cnt + 1]);
+        hal.scheduler->delay(10);
+        for (int i = 0; i < 3; i++) {
+            recv_buf[i] = uart->read();
+        }
+        // if failed to receive response code, you can start over from 0xBB.
+        if ((recv_buf[0] ^ recv_buf[1]) != recv_buf[2]) {
+            UPFLOW_INIT_FAILED;
+            return;
+        }
+        cfg_cnt += 2;
+    }
+
+    // (0xDD)End configuration
+    uart->write((uint8_t)0xDD);
+
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "UPFLOW: initialized Done!");
```

Comment:
Don't need this message.

Reply:
I apologize, this message is what I set up to observe initialization behavior in the ground station. If initialization is successful, the GCS will display _UPFLOW: initialized Done!_. If initialization fails, it assumes the user is using an LC302-3C, and the GCS will display _UPFLOW: Init Failed, Using LC302-3C_ and exit the initialization process.

If this message prompt is unnecessary, then it can be removed along with the _UPFLOW_INIT_FAILED_ macro.